services/meta/write_authorizer.go

```diff
@@ -19,11 +19,25 @@ func NewWriteAuthorizer(c *Client) *WriteAuthorizer {
 // AuthorizeWrite returns nil if the user has permission to write to the database.
 func (a WriteAuthorizer) AuthorizeWrite(username, database string) error {
 	u, err := a.Client.User(username)
-	if err != nil || u == nil || !u.AuthorizeDatabase(influxql.WritePrivilege, database) {
+	if err != nil || u == nil {
 		return &ErrAuthorize{
 			Database: database,
 			Message:  fmt.Sprintf("%s not authorized to write to %s", username, database),
 		}
 	}
+	switch user := u.(type) {
```

Comment:
Sure - there is only one OSS implementation of the `User` interface, and the OSS WriteAuthorizer only works with the OSS UserInfo.

There is a similar tight coupling between the Enterprise WriteAuthorizer and Enterprise UserInfo in closed-source code.

The tight coupling was already implicitly present, this PR makes the coupling explicit instead of doing major changes to try to avoid it.

I'll add a comment explaining this where we added the type switches.

Reply:
Done